src/scipp/core/dimensions.py

```diff
@@ -49,8 +50,21 @@ def _rename_dataset(ds: Dataset, dims_dict: dict = None, **names) -> Dataset:
        - using keyword arguments, e.g. rename(x='a', y='b')
     """
     renaming_dict = {**({} if dims_dict is None else dims_dict), **names}
-    out = Dataset()
+    ds_from_items = Dataset()
     for key, item in ds.items():
         dims_dict = {old: new for old, new in renaming_dict.items() if old in item.dims}
-        out[key] = _rename_data_array(ds[key], dims_dict=dims_dict)
-    return out
+        # Rename data and attrs but not coords, to avoid the cost of renaming them a
+        # second time below.
```

Comment:
> Note that this is basically for free. Minimizing risk of bugs would be more important.

yes i came to the same conclusion.

> If you remove this, can you make a generic _rename_map that you can use to avoid the duplication between renaming the map of items and the map of coords (below)?

I tried doing this but it got a bit messy because in one case i want to call `_rename_data_array` and in the other `_rename_variable`. But i can have another go...

Reply:
> What about conflicts between coords and attrs? It seems they cannot be detected any more like this?

what conflicts are you refering to exactly? I thought that you could have coords and attrs with the same name in a DataArray, because preventing that is difficult?

Reply:
Yes, but we should not create such situations deliberately? Also, you are protecting against it in `_rename_data_array`, so why not for datasets as well, to be consistent?

Reply:
> I'm not cross-checking between coords and attrs

I think we have to. Should we just check something like `if new_name in da.meta: raise ...`?

> Unless I was meant to demote coords to attrs (a bit like in transform_coords) after renaming, in which case I misunderstood the issue...

No, we don't want to demote.